### PR TITLE
docs: README refresh, add cli-http-api-core diagram

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,8 +9,6 @@
 [![GoDoc](https://godoc.org/github.com/ipfs/go-ipfs?status.svg)](https://godoc.org/github.com/ipfs/go-ipfs)
 [![Build Status](https://travis-ci.org/ipfs/go-ipfs.svg?branch=master)](https://travis-ci.org/ipfs/go-ipfs)
 
-> IPFS implementation in Go
```
Comment (Stebalien) [inline README.md]: I'd leave this (unless there's a good reason to remove it).

File: README.md [hunk 2]
```diff
@@ -317,41 +318,72 @@ Stop the running container:
...
-If you believe you've found a bug, check the [issues list](https://github.com/ipfs/go-ipfs/issues)
-and, if you don't see your problem there, either come talk to us on IRC (freenode #ipfs) or
-file an issue of your own!
+If you believe you've found a bug, check the [issues list](https://github.com/ipfs/go-ipfs/issues) and, if you don't see your problem there, either come talk to us on IRC (freenode #ipfs) or file an issue of your own!
 
-## Contributing
+## Development
 
-We ❤️ all [our contributors](docs/AUTHORS); this project wouldn’t be what it is without you! If you want to help out, please see [Contribute.md](contribute.md).
+This is a simple description of where the codebase stands. There are multiple subpackages:
```
Comment (Stebalien) [inline README.md]: This is all out of date. We should probably just remove it.

File: README.md [hunk 3]
```diff
@@ -317,41 +318,72 @@ Stop the running container:
...
+- `swarm` - connection multiplexing, many peers and many transports
+- `util` - various utilities
 
-### Want to hack on IPFS?
+Some places to get you started on the codebase:
 
-[![](https://cdn.rawgit.com/jbenet/contribute-ipfs-gif/master/img/contribute.gif)](https://github.com/ipfs/community/blob/master/contributing.md)
+- Main file: [cmd/ipfs/main.go](https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/main.go)
+- CLI Commands: [core/commands/](https://github.com/ipfs/go-ipfs/tree/master/core/commands)
+- Bitswap (the data trading engine): [exchange/bitswap/](https://github.com/ipfs/go-ipfs/tree/master/exchange/bitswap)
```
Comment (Stebalien) [inline README.md]: Moved to https://github.com/ipfs/go-bitswap

File: README.md (hunk 1 above)
Comment (diasdavid) [reply README.md]: Isn't it obvious?

Comment (Stebalien) [reply README.md]: I figure it can't hurt but you're right, it doesn't *really* have to be there.

**Pull request merged**